src/ImageIO.jl

```diff
@@ -33,11 +33,11 @@ end
 
 function load(f::File{DataFormat{:PNG}}; kwargs...)
     data = Base.invokelatest(checked_import(idPNGFiles).load, f.filename, kwargs...)
-    return convert(canonical_type(f, data), data)
+    return Base.invokelatest(convert, canonical_type(f, data), data)
```

Comment:
I think we can shave this `invokelatest` overhead by conditionally calling `convert` when `!isa(data, canonical_type(f, data))`